src/easier68k/assembler/assembler.py

```diff
@@ -37,10 +46,93 @@ def parse(text: str):  # should return a list file and errors/warnings eventuall
             else:
                 labels[label] = (line_index, label_contents)
                 if get_opcode(stripped) == 'EQU':
-                    equates[label] = parse_literal(strip_opcode(stripped))
+                    equates[label] = strip_opcode(stripped)
 
     # --- PART 2: process operations for sizing and lay out memory ---
 
-    for m in opcodes.__all__:
-        module = sys.modules['easier68k.core.opcodes.{}'.format(m)]
-        # print(module)
+    to_return = ListFile()
+    current_memory_location = 0x00000000
+    label_addresses = {}  # Stores all of the label memory locations
+
+    for line_index, stripped in for_line_stripped_comments(text):
+        opcode = get_opcode(stripped).upper()
+
+        # Equates have already been processed, skip them
+        # (this idea could be expanded for more preprocessor directives)
+        if opcode == 'EQU':
+            continue
+
+        contents = strip_opcode(stripped)
+
+        # Replace all substitutions in the current line with their corresponding values
+        for equate in equates.items():
+            contents = contents.replace(equate[0], equate[1])
+
+        if opcode == 'ORG':  # This will shift our current memory location, it's a special case
+            parsed = parse_literal(contents)
+            # TODO: Max/min memory location?
+            current_memory_location = int.from_bytes(parsed, 'big')
+            continue
+
+        if has_label(stripped):
+            label = get_label(stripped)
+            label_addresses[label] = current_memory_location
+            to_return.define_symbol(label, current_memory_location)
+
+        # We don't know this opcode, there's no module for it
+        if opcode.lower() not in opcodes.__all__:
```

Comment:
I don't think this is how `__all__` should be used, since the point of `__all__` is to be used for importing, so it would also import the base `OpCode` class.

Instead, I think you should make a list of paths to the opcodes, then use a method in each of those modules to evaluate if the value is valid for the opcode

Reply:
I think I see the start of that in your other PR, I'll pull from that and add a method checking if it matches.